src/components/app/data/hooks/useEnterpriseFeatures.test.jsx

```diff
@@ -1,65 +1,108 @@
 
+const mockBFFDashboardData = {
+  enterpriseCustomer: mockEnterpriseCustomer,
+  allLinkedEnterpriseCustomerUsers: [],
+  enterpriseFeatures: {
+    isBFFEnabled: true,
+  },
+  shouldUpdateActiveEnterpriseCustomerUser: false,
+  enterpriseCustomerUserSubsidies: {
+    subscriptions: {
+      customerAgreement: {},
+      subscriptionLicenses: [],
+      subscriptionLicensesByStatus: {
+        activated: [],
+        assigned: [],
+        expired: [],
+        revoked: [],
+      },
+    },
+  },
+  enterpriseCourseEnrollments: [],
+  errors: [],
+  warnings: [],
+};
+
+const mockExpectedEnterpriseFeatures = (isMatchedBFFRoute) => (isMatchedBFFRoute
+  ? mockBFFDashboardData.enterpriseFeatures
+  : mockEnterpriseLearnerData.enterpriseFeatures);
 describe('useEnterpriseFeatures', () => {
-  const Wrapper = ({ children }) => (
+  const Wrapper = ({ initialEntries = [], children }) => (
     <QueryClientProvider client={queryClient()}>
-      <AppContext.Provider value={{ authenticatedUser: mockAuthenticatedUser }}>
-        {children}
-      </AppContext.Provider>
+      <MemoryRouter initialEntries={initialEntries}>
+        <AppContext.Provider value={{ authenticatedUser: mockAuthenticatedUser }}>
+          <Routes>
+            <Route path=":enterpriseSlug" element={children} />
+            <Route path=":enterpriseSlug/search" element={children} />
+          </Routes>
+          {children}
+        </AppContext.Provider>
+      </MemoryRouter>
     </QueryClientProvider>
   );
 
   beforeEach(() => {
     jest.clearAllMocks();
+    fetchEnterpriseLearnerDashboard.mockResolvedValue(mockBFFDashboardData);
     fetchEnterpriseLearnerData.mockResolvedValue(mockEnterpriseLearnerData);
   });
 
-  it.each([
-    { hasQueryOptions: false },
-    { hasQueryOptions: true },
-  ])('should return enterprise features correctly (%s)', async ({ hasQueryOptions }) => {
+  it.each(generateTestPermutations({
+    isMatchedBFFRoute: [false, true],
+    hasCustomSelect: [false, true],
+  }))('should return enterprise features correctly (%s)', async ({
+    isMatchedBFFRoute,
+    hasCustomSelect,
+  }) => {
     const mockSelect = jest.fn(data => data.transformed);
+    const initialEntries = isMatchedBFFRoute ? ['/test-enterprise'] : ['/test-enterprise/search'];
+    const enterpriseLearnerHookArgs = hasCustomSelect ? { select: mockSelect } : {};
     const { result, waitForNextUpdate } = renderHook(
-      () => {
-        if (hasQueryOptions) {
-          return useEnterpriseFeatures({ select: mockSelect });
-        }
-        return useEnterpriseFeatures();
+      () => (useEnterpriseFeatures(enterpriseLearnerHookArgs)),
+      {
+        wrapper: ({ children }) => (
+          <Wrapper initialEntries={initialEntries}>
+            {children}
+          </Wrapper>
+        ),
       },
-      { wrapper: Wrapper },
     );
     await waitForNextUpdate();
-
-    if (hasQueryOptions) {
-      expect(mockSelect).toHaveBeenCalledWith({
```

Comment:
Was the check on the `expect(mockSelect).toHaveBeenCalledWith` intentionally removed?

Reply:
Yep, I reintroduced the field as `hasCustomSelect` but I brought the logic back originally removed to validate the values passed in 👍🏽